<launch>
    <!-- Configuration Settings -->
    <arg name="marker_size" value="15.0"/>
    <arg name="max_new_marker_error" value="0.05"/>
    <arg name="max_track_error" value="0.05"/>
    <arg name="cam_image_topic" value="/usb_cam/image_raw"/>
    <arg name="cam_info_topic" value="/usb_cam/camera_info"/>
    <arg name="output_frame" value="/usb_cam"/>
  
    <!-- RVIZ Node -->
    <node name="rviz" pkg="rviz" type="rviz" output="screen"/>

    <!-- Camera Node -->
    <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="log">
        <param name="video_device" value="/dev/video0"/>
        <param name="image_width" value="1280"/>
        <param name="image_height" value="720"/>
        <param name="pixel_format" value="yuyv"/>
        <param name="camera_frame_id" value="usb_cam"/>
        <param name="io_method" value="mmap"/>
    </node>

    <!-- AR Tracking Node -->
    <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" output="log">
	    <param name="marker_size" type="double" value="$(arg marker_size)"/>
	    <param name="max_new_marker_error" type="double" value="$(arg max_new_marker_error)"/>
	    <param name="max_track_error" type="double" value="$(arg max_track_error)"/>
	    <param name="output_frame" type="string" value="$(arg output_frame)"/>
	    <remap from="camera_image" to="$(arg cam_image_topic)"/>
	    <remap from="camera_info" to="$(arg cam_info_topic)"/>
    </node>
</launch>
